From a68de780fe7aba8713cd1d3792b4c8fe4d585f18 Mon Sep 17 00:00:00 2001
From: ARATA Mizuki <minorinoki@gmail.com>
Date: Sun, 12 Jul 2015 21:49:13 +0900
Subject: [PATCH 1/5] Fix wxMBConvUTF8::FromWChar.

The function was incorrectly converting surrogate pairs if wchar_t is UTF-16.
---
 src/common/strconv.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/common/strconv.cpp b/src/common/strconv.cpp
index a0898cf..8894c37 100644
--- a/src/common/strconv.cpp
+++ b/src/common/strconv.cpp
@@ -1400,6 +1400,8 @@ size_t wxMBConvUTF8::FromWChar(char *buf, size_t n,
         // cast is ok for WC_UTF16
         size_t pa = decode_utf16((const wxUint16 *)psz, cc);
         psz += (pa == wxCONV_FAILED) ? 1 : pa;
+        if (pa == 2 && !isNulTerminated)
+            srcLen--;
 #else
         cc = (*psz++) & 0x7fffffff;
 #endif
-- 
2.4.5

